src/PIL/TiffImagePlugin.py

```diff
@@ -2101,8 +2108,16 @@ def skipIFDs(self) -> None:
             num_tags = self.readShort()
             self.f.seek(num_tags * 12, os.SEEK_CUR)
 
-    def write(self, data: bytes) -> int | None:
-        return self.f.write(data)
+    if sys.version_info >= (3, 12):
+        from collections.abc import Buffer
+
+        def write(self, data: Buffer, /) -> int:
+            return self.f.write(data)
+
+    else:
+
+        def write(self, data: Any, /) -> int:
+            return self.f.write(data)
```

Comment:
Perhaps something like this?
```suggestion
    if sys.version_info >= (3, 12):
        from collections.abc import Buffer
    else:
        Buffer = Any

    def write(self, data: Buffer, /) -> int:
        return self.f.write(data)
```

Reply:
Sure. I've actually moved `Buffer` into `src/PIL/_typing.py` so that this change can occur for GifImagePlugin cleanly as well.